PryntTrimmerView/Classes/Parents/AssetVideoScrollView.swift

```diff
@@ -11,117 +11,182 @@ import UIKit
+            return
+        }
+        
+        delegate?.contentOffsetDidChange()
+    }
+    
     override func layoutSubviews() {
         super.layoutSubviews()
-        contentSize = contentView.bounds.size
+        
+        defer {
+            self.lastWidth = bounds.width
+        }
+       
         guard let duration = duration, let thumbnailFrameAspectRatio = thumbnailFrameAspectRatio else {
             return
         }
+        
+        guard let lastWidth = lastWidth else {
+            recalculateThumbnailTimes(for: duration, thumbnailFrameAspectRatio: thumbnailFrameAspectRatio)
+            return
+        }
+        
+        guard lastWidth != bounds.width else {
+            return
+        }
+        
+        lastContentOffset = collectionView.contentOffset.x
         recalculateThumbnailTimes(for: duration, thumbnailFrameAspectRatio: thumbnailFrameAspectRatio)
+        recalculateContentOffset(lastWidth)
+    }
+    
+    private func recalculateContentOffset(_ lastWidth: CGFloat) {
+        let ratio = bounds.width / lastWidth
+        guard lastContentOffset > horizontalInset else {
+            return
+        }
+        
+        collectionView.contentOffset = CGPoint(x: (lastContentOffset - horizontalInset) * ratio + horizontalInset,
+                                               y: collectionView.contentOffset.y)
     }
 
     internal func recalculateThumbnailTimes(for duration: TimeInterval, thumbnailFrameAspectRatio: CGFloat) {
+        let thumbnailSize = getThumbnailFrameSize(for: thumbnailFrameAspectRatio)
         guard
-            let thumbnailSize = getThumbnailFrameSize(for: thumbnailFrameAspectRatio),
             thumbnailSize.height.isNormal,
             thumbnailSize.width.isNormal else {
-            return
+                return
         }
         
+        self.thumbnailSize = thumbnailSize
         self.thumbnailFrameAspectRatio = thumbnailFrameAspectRatio
         self.duration = duration
         
-        removeFormerThumbnails()
-        let newContentSize = frame.size  // setContentSize(for: asset)
-        let visibleThumbnailsCount = Int(ceil(frame.width / thumbnailSize.width))
-        let thumbnailCount =  thumbnailSize.width > 0 ? Int(ceil(newContentSize.width / thumbnailSize.width)) : 0
-        addThumbnailViews(thumbnailCount, size: thumbnailSize)
-        let thumbnailTimes = getThumbnailTimes(for: duration, numberOfThumbnails: thumbnailCount)
-        generateImages(at: thumbnailTimes, with: thumbnailSize, visibleThumnails: thumbnailCount)
+        let thumbnailCount =  getThumbnailCount(for: duration)
+        thumbnailTimes = getThumbnailTimes(for: duration, numberOfThumbnails: thumbnailCount)
+        self.collectionView.reloadData()
+        self.collectionView.performBatchUpdates(nil) { _ in
+            self.delegate?.didUpdateDimensions()
+        }
+    }
+    
+    private func getThumbnailCount(for duration: TimeInterval) -> Int {
```

Comment:
In iOS it is common not to start functions or vars with "get". Just "thumbnailCount" is the best practice.

Reply:
used an existing func, but I'l change 👍